src/main/java/io/vertx/core/eventbus/impl/clustered/ConnectionHolder.java

```diff
@@ -61,32 +62,35 @@
     client = new NetClientImpl(eventBus.vertx(), clientOptions, false);
   }
 
-  synchronized void connect() {
-    if (connected) {
-      throw new IllegalStateException("Already connected");
-    }
-    client.connect(serverID.port, serverID.host, res -> {
-      if (res.succeeded()) {
-        connected(res.result());
-      } else {
-        log.warn("Connecting to server " + serverID + " failed", res.cause());
-        close(res.cause());
+  void connect() {
+    synchronized (this) {
+      if (connected) {
+        throw new IllegalStateException("Already connected");
       }
-    });
+    }
+    eventBus.vertx().getClusterManager().getNodeInfo(remoteNodeId)
+      .flatMap(nodeInfo -> nodeInfo != null ? Future.succeededFuture(nodeInfo) : Future.failedFuture("Not a member of the cluster"))
+      .map(NodeInfo::getAddress)
+      .flatMap(address -> client.connect(address.getPort(), address.getHost()))
+      .onSuccess(this::connected)
```

Comment:
use `onComplete` instead of `onSuccess`/`onFailure`

Reply:
Done